magenta/models/rl_tuner/note_rnn_loader.py

```diff
@@ -0,0 +1,423 @@
+          else:
+            (self.logits, self.state_tensor) = run_network_on_melody(
+                self.melody_sequence, self.lengths, self.initial_state)
+            self.softmax = tf.nn.softmax(self.logits)
+
+          self.run_network_on_melody = run_network_on_melody
+
+        if self.training_file_list is not None:
+          # Does not recreate the model architecture but rather uses it to feed
+          # data from the training queue through the model.
+          with tf.variable_scope(self.scope, reuse=True):
+            zero_state = self.cell.zero_state(
+                batch_size=self.hparams.batch_size, dtype=tf.float32)
+
+            if self.softmax_within_graph:
+              (self.train_softmax, self.train_state) = run_network_on_melody(
+                  self.train_sequence, self.train_lengths, zero_state)
+            else:
+              (self.train_logits, self.train_state) = run_network_on_melody(
+                  self.train_sequence, self.train_lengths, zero_state)
+              self.train_softmax = tf.nn.softmax(self.train_logits)
+
+  def restore_vars_from_checkpoint(self, checkpoint_dir):
+    """Loads model weights from a saved checkpoint.
+
+    Args:
+      checkpoint_dir: Directory which contains a saved checkpoint of the
+        model.
+    """
+    tf.logging.info('Restoring variables from checkpoint')
+
+    var_dict = self.get_variable_name_dict()
+    with self.graph.as_default():
+      saver = tf.train.Saver(var_list=var_dict)
+
+    tf.logging.info('Checkpoint dir: %s', checkpoint_dir)
+    checkpoint_file = tf.train.latest_checkpoint(checkpoint_dir)
+    if checkpoint_file is None:
+      tf.logging.warn("Can't find checkpoint file, using backup, which is %s", self.backup_checkpoint_file)
+      checkpoint_file = self.backup_checkpoint_file
+    tf.logging.info('Checkpoint file: %s', checkpoint_file)
+
+    saver.restore(self.session, checkpoint_file)
+
+  def load_primer(self):
+    """Loads default MIDI primer file.
+
+    Also assigns thesteps per bar of this file to be the model's defaults.
+    """
+
+    if not os.path.exists(self.midi_primer):
+      tf.logging.warn('ERROR! No such primer file exists! %s', self.midi_primer)
+      return
+
+    self.primer_sequence = midi_io.midi_file_to_sequence_proto(self.midi_primer)
+    quantized_seq = sequences_lib.QuantizedSequence()
+    quantized_seq.from_note_sequence(self.primer_sequence,
+                                     steps_per_quarter=4)
+    extracted_melodies, _ = melodies_lib.extract_melodies(quantized_seq,
+                                                          min_bars=0,
+                                                          min_unique_pitches=1)
+    self.primer = extracted_melodies[0]
+    self.steps_per_bar = self.primer.steps_per_bar
+
+  def prime_model(self, suppress_output=False):
+    """Primes the model with its default midi primer."""
+    with self.graph.as_default():
+      if not suppress_output:
```

Comment:
here and elsewhere, rather than having a suppress_output flag, just log this at a lower logging level (e.g., debug). You can add a flag to your main entry point for controlling the log level. Here's an example: https://github.com/tensorflow/magenta/blob/master/magenta/models/melody_rnn/melody_rnn_train.py#L177